src/test/java/org/springframework/data/redis/connection/lettuce/LettuceConvertersUnitTests.java

```diff
@@ -252,4 +255,18 @@ void convertsExpirationToGetExPXAT() {
 		assertThatCommandArgument(LettuceConverters.toGetExArgs(Expiration.unixTimestamp(10, TimeUnit.MILLISECONDS)))
 				.isEqualTo(new GetExArgs().pxAt(10));
 	}
+
+	@Test
+	void sentinelConfigurationWithAuth() {
+		RedisSentinelConfiguration sentinelConfiguration = new RedisSentinelConfiguration()
+				.master(MASTER_NAME)
+				.sentinel("127.0.0.1", 26379)
+				.sentinel("127.0.0.1", 26380);
+		sentinelConfiguration.setSentinelUsername("admin");
+		sentinelConfiguration.setSentinelPassword(RedisPassword.of("88888888-8x8-getting-creative-now"));
+		sentinelConfiguration.setUsername("app");
+		sentinelConfiguration.setPassword(RedisPassword.of("88888888-8x8-getting-creative-now"));
+		RedisURI redisURI = LettuceConverters.sentinelConfigurationToRedisURI(sentinelConfiguration);
+		assertThat(redisURI.toString()).isEqualTo("redis-sentinel://app:*********************************@127.0.0.1,127.0.0.1:26380?sentinelMasterId=mymaster");
```

Comment:
Please rewrite this check to inspect the `password` property of each sentinel URI (`RedisURI.getSentinels()`). The `toString` check performs a very strict test on unrelated internals and also the rendering of the password isn't part of any API contract.